### PR TITLE
Meetup Page is created

### PR DESCRIPTION
**Pull request by madhusudhan1234**
### Summary of the changes

Meetup page is created and reference is taken from [Here](http://developers-nepal.github.io/ktmjs/)
HTML Markup is added

### Why we need this
- [x] Meetup information can be seen from the website

### Main Changes
- [x] Meetup page markup and css added
- [x] meetup.yml is added and adjusted required data
- [x] Font is changed
- [x] Used Real data for the meetup
- [x] React Js is used 
- [x] User Information are in now json file

### CODE REVIEW

File: _sass/_layouts/_home.scss [hunk 1]
```diff
@@ -1,6 +1,6 @@
 body {
   background: $body-background-color;
-  font-family: 'Vollkorn', serif;
+  font-family: 'Source Sans Pro', sans-serif;
```
Comment (zoras) [inline _sass/_layouts/_home.scss]: Why is the main font changed?

Comment (madhusudhan1234) [reply _sass/_layouts/_home.scss]: While Using ```'Vollkorn', serif;``` 1 is seems like I, And I searched better font ```'Source Sans Pro', sans-serif;``` is looks good for me. That's why I changed the font. 

screen shot 2018-06-03 at 8 51 12 pm

File: _includes/nav-links.html [hunk 2]
```diff
@@ -1,11 +1,13 @@
 <a class="navbar-item blog {% if (page.url contains 'blog' %} active-tab {% endif %}"
   href="/blog">Blog</a>
+<a class="navbar-item meetup {% if page.url contains 'meetup' %} active-tab {% endif %}"
+  href="/meetup">Meetup</a>
 <a class="navbar-item learn {% if page.url contains 'learn' %} active-tab {% endif %}" id="learn"
   href="/learn">Learn</a>
 <a class="navbar-item meetup"
```
Comment (zoras) [inline _includes/nav-links.html]: Remove this meetup tag as it's duplicated

File: _includes/nav-links.html [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
   href="https://www.meetup.com/{{ site.meetup_username }}" target="_blank">Meetup</a>
 <a class="navbar-item slack"
-  href="https://{{ site.slack_username }}.slack.com/" target="_blank">Chat</a>
+   href="https://{{ site.slack_username }}.slack.com/" target="_blank">Chat</a>
```
Comment (zoras) [inline _includes/nav-links.html]: Revert the addition of extra space

File: _data/meetup.yml [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+      github_username: ''
+      designation: Fun-loving Adventurous Free-spirited Determined
+      slide_link: null
+      video_link: null
+      demo_link: null
+    - id: 2
+      title: Organizing CSS
+      description: null
+      speaker: Sanjeev Praja
+      twitter_username: null
```
Comment (zoras) [inline _data/meetup.yml]: https://twitter.com/sanjeevpraja
https://github.com/SanjeevPraja

File: _data/meetup.yml [hunk 5]
```diff
@@ -0,0 +1,406 @@
...
+    link: 'http://www.leapfrog.academy/'
+    logo: 'http://www.leapfrog.academy/themes/website/assets/img/logo_main.png'
+    name: Leapfrog Academy
+  sessions:
+    - id: 1
+      title: Working as a Junior Developer
+      description: null
+      speaker: Madhuri Shrestha
+      twitter_username: MadzzSrestha
+      github_username: ''
```
Comment (zoras) [inline _data/meetup.yml]: https://github.com/MadhuriShres

File: _includes/nav-links.html (hunk 2 above)
Comment (madhusudhan1234) [reply _includes/nav-links.html]: oh ! okay seems it came during resolve conflict.

File: _includes/nav-links.html (hunk 3 above)
Comment (madhusudhan1234) [reply _includes/nav-links.html]: okay dai

**Comment by RohitRox** [comment]
This is great work. Really like the new meetup page. 
My feedback:

- The prev and next arrow sometimes get hidden(gets overlapped by the footer)
- Can we increase the size of the arrows?
- Would be nice to have links to videos/slides if available
- The gap between session number, red border and title looks a bit large, may be we should play with margins/padding and see what looks the best

**Comment by madhusudhan1234** [comment]
Hi @josisusan dai, Thank you for the response. As per your feedback, I would like to list following action items.

- [ ] The prev and next arrows are hidden by footer at the time of scrolling to the bottom. what we can do for the fix is we can add one class at the time of scrolling and reduce the top padding of the fixed section. That should fix the issue.
- [ ] Yes we can increase the size of the arrow
- [ ] Yes, there is videos/slides links are already in data we will show that information.
- [ ] okay we can reduce padding and make better for title and session number gap

@zoras dai, @josisusan dai If above action items are okay to you. I will fix those issue and push code for review.

Note: I am trying to fix above Hound 6 violations, and not able to fix yet.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    "build": "webpack --mode production"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/madhusudhan1234/rubynepal.github.io.git"
+  },
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/madhusudhan1234/rubynepal.github.io/issues"
```
Comment (josisusan) [inline package.json]: Better to change this link to `https://github.com/rubynepal/rubynepal.github.io/issues`

File: package.json [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  "private": true,
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "webpack --mode development",
+    "watch": "webpack --mode development --watch",
+    "build": "webpack --mode production"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/madhusudhan1234/rubynepal.github.io.git"
```
Comment (josisusan) [inline package.json]: same here 😄

File: package.json (hunk 6 above)
Comment (madhusudhan1234) [reply package.json]: okay dai, I will change 👍

File: js/meetup.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './meetup/App';
+
+window.addEventListener('scroll', function() {
```
Comment (RohitRox) [inline js/meetup.js]: I think this function should be in `componentDidMount` of  App component and `SECTION` (should be lowercase) should be used via use of refs.

File: js/meetup/App.js [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+    super(props);
+
+    this.state ={
+      selectedEpisode: this.latestEpisode(),
+      firstEpisode: 0,
+      messageForNotFound: '',
+    };
+
+    this.onStepChange = this.onStepChange.bind(this);
+    this.latestEpisode = this.latestEpisode.bind(this);
```
Comment (RohitRox) [inline js/meetup/App.js]: Minor refactor, define function with arrow head `onStepChange  = () => { ... }` to get rid of these ugly binds.

**Comment by RohitRox** [comment]
Is there a way we can run hound checks in local? so that we can fix issues before we push or even have a auto correct and pre-push hook.  Hound CI comments are polluting PRs.

File: js/meetup/App.js [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+  }
+}
+
+function SlideAndVideo(props) {
+  const slide_link =  props.session.slide_link;
+  const video_link =  props.session.video_link;
+  const demo_link =  props.session.demo_link;
+  if (props.session && (slide_link || video_link || demo_link)) {
+    return (<div className="slide-and-video">
+      {slide_link && <a href={slide_link} target="_blank"><i className="fas fa-sliders-h"></i>  Slide</a>} &nbsp;&nbsp;
```
Comment (RohitRox) [inline js/meetup/App.js]: Slide should be in plural form, Slides

**Comment by RohitRox** [comment]
Looks good except my few comments 👍 Cheers

File: js/meetup.js (hunk 8 above)
Comment (madhusudhan1234) [reply js/meetup.js]: okay, @RohitRox  dai I will do that.

File: js/meetup/App.js (hunk 9 above)
Comment (madhusudhan1234) [reply js/meetup/App.js]: oh! Yeah, that would be better. I will remove those ugly binds :)

**Comment by madhusudhan1234** [comment]
> Is there a way we can run hound checks in local? so that we can fix issues before we push or even have a auto correct and pre-push hook. Hound CI comments are polluting PRs.

> Yes @RohitRox dai, there is an option to run hound locally but in a local environment, there are no any Hound's violations. It is quite strange.

File: js/meetup/App.js (hunk 10 above)
Comment (madhusudhan1234) [reply js/meetup/App.js]: okay, I will make the plural.

**Comment by madhusudhan1234** [comment]
> Looks good except my few comments 👍 Cheers

Thank You @RohitRox dai, The remaining part is to add URL for each meetup and left-right keyboard binding. I will change the code as per your review suggestion in the same commit.

**Comment by RohitRox** [comment]
> > Is there a way we can run hound checks in local? so that we can fix issues before we push or even have a auto correct and pre-push hook. Hound CI comments are polluting PRs.
> 
> Yes @RohitRox dai, there is an option to run hound locally but in a local environment, there are no any Hound's violations. It is quite strange.

May be @zoras dai can help

File: js/meetup.js [hunk 11]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (zoras) [reply js/meetup.js]: @madhusudhan1234 see https://github.com/eslint/eslint/issues/5552#issuecomment-195544822
https://eslint.org/docs/user-guide/configuring#specifying-parser-options

**Comment by zoras** [comment]
> > > Is there a way we can run hound checks in local? so that we can fix issues before we push or even have a auto correct and pre-push hook. Hound CI comments are polluting PRs.
> > 
> > 
> > Yes @RohitRox dai, there is an option to run hound locally but in a local environment, there are no any Hound's violations. It is quite strange.
> 
> May be @zoras dai can help

The @houndci-bot errors are now fixed by adding custom `.eslintrc.json` file

**Comment by josisusan** [comment]
👍  @madhusudhan1234  @zoras

**Pull request merged**